File: 3536-maximum-product-of-two-digits/3536-maximum-product-of-two-digits.go
func maxProduct(n int) int {
    temp := []int{}
    m := 1
    for n > 0 {
        d := n % (m * 10)
        temp = append(temp, d)
        n /= 10
    }

    sort.Ints(temp)
    
    return temp[len(temp)-1] * temp[len(temp)-2]
}

// math, sorting
// time: O(klog(k))
// space: O(k)

File: 3536-maximum-product-of-two-digits/3536-maximum-product-of-two-digits.py
class Solution:
    def maxProduct(self, n: int) -> int:
        temp = []
        
        while n > 0:
            digit = n % 10
            temp.append(digit)
            n //= 10
        
        temp.sort()

        return temp[len(temp) - 1] * temp[len(temp) - 2]

# math, sorting
# time: O(klog(k))
# space: O(k)